src/main/kotlin/lotto/domain/LottoGame.kt

```diff
@@ -1,20 +1,21 @@
 package lotto.domain
 
-const val PRICE_OF_LOTTO = 1000
-
 class LottoGame(val lottoList: List<Lotto>) {
-    val lottoPrizeStatics = LottoPrizeStatics()
-
-    constructor(gameMoneyString: String) : this(
-        LottoGenerator.createAutoLottoList(gameMoneyString)
-    )
 
-    fun execute(prizeNumberString: String, bonusNumberString: String) {
-        val winningLotto = WinningLotto(prizeNumberString, bonusNumberString)
-        checkMatch(winningLotto)
+    fun execute(prizeNumbers: String, bonusNumberInput: String): LottoGameResult {
+        val prizeLotto = Lotto.from(prizeNumbers) ?: return LottoGameResult.InvalidPrizeLotto
+        if (!LOTTO_NUMBER_REGULAR_EXPRESSION.matches(bonusNumberInput)) return LottoGameResult.InvalidBonusNumber
+        val bonusNumber: LottoNumber = LottoNumber.from(bonusNumberInput)
+        if (prizeLotto.isContainNumber(bonusNumber)) return LottoGameResult.IsContainBonusNumber
+        val winningLotto = WinningLotto(prizeLotto, bonusNumber)
+        val prizeStatics = LottoPrizeStatics(winningLotto, lottoList)
+        return LottoGameResult.Success(prizeNumbers, bonusNumber, prizeStatics)
     }
 
-    private fun checkMatch(winningLotto: WinningLotto) {
-        lottoPrizeStatics.calculateResult(winningLotto, lottoList)
+    companion object {
+        fun of(gameMoney: LottoGameMoney, manualLottos: List<Lotto> = listOf()): LottoGame {
+            val autoLottoCount = gameMoney.getCountOfGame() - manualLottos.size
+            return LottoGame(manualLottos.plus(LottoGenerator.createAutoLottoList(autoLottoCount)))
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
            return LottoGame(manualLottos + LottoGenerator.createAutoLottoList(autoLottoCount))
```

Reply:
https://github.com/next-step/kotlin-lotto/commit/186f13640a87e2d10812845acf3c0e52b9672788

수정 완료했습니다.